### PR TITLE
Install the correct certificates for filebeat

### PR DESCRIPTION
**Pull request by arturo-seijas**
Applicable spec: <link>

### Overview

Follow up #23 

Also fixes filebeat startup

### Rationale

The list of certificates is not correctly matches by csr. Using the method provided by the lib should handle the decoding of this field while simplifying the code

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -59,14 +59,9 @@ def state(self) -> State | None:
...
-            certificates_relation = self.model.get_relation(certificates_observer.RELATION_NAME)
-            certificates_relation_data = (
-                certificates_relation.data[certificates_relation.app]
-                if certificates_relation
-                else {}
-            )
+            certificates = self.certificates.certificates.get_provider_certificates()
             return State.from_charm(
-                self, opensearch_relation_data, certificates_relation_data, self.certificates.csr
+                self, opensearch_relation_data, certificates, self.certificates.csr.decode("utf-8")
```
Comment (amandahla) [inline src/charm.py:64]: Any chance of csr being empty/None?

**Review by amandahla — DISMISSED** [review]

**Review by javierdelapuente — APPROVED** [review]

**Review by nrobinaubertin — DISMISSED** [review]

**Review by Thanhphan1147 — APPROVED** [review]
LGTM!

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -20,11 +20,19 @@
...
 
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy(model: Model, application: Application):
+async def test_filebeat_ok(model: Model, application: Application):
     """Deploy the charm together with related charms.
 
-    Assert on the unit status.
+    Assert: the filebeat config is valid.
     """
+    assert False
```
Comment (cbartz) [inline tests/integration/test_charm.py:28]: I guess the tests is still WIP?

Comment (arturo-seijas) [reply tests/integration/test_charm.py]: Oops. Fixed. Thanks :)

**Review by cbartz — APPROVED** [review]

**Review by javierdelapuente — APPROVED** [review]

**Review by Thanhphan1147 — APPROVED** [review]

**Review by cbartz — DISMISSED** [review]

**Review by amandahla — DISMISSED** [review]

**Review by cbartz — APPROVED** [review]

**Review by amandahla — APPROVED** [review]

**Pull request merged**